x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/crawler_domains_logic.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { Meta } from '../../../../../common/types';
+import { DEFAULT_META } from '../../../shared/constants';
+import { flashAPIErrors } from '../../../shared/flash_messages';
+import { HttpLogic } from '../../../shared/http';
+import { updateMetaPageIndex } from '../../../shared/table_pagination';
+import { EngineLogic } from '../engine';
+
+import {
+  CrawlerData,
+  CrawlerDataFromServer,
+  CrawlerDomain,
+  CrawlerDomainFromServer,
+} from './types';
+import { crawlerDataServerToClient, crawlerDomainServerToClient } from './utils';
+
+export interface CrawlerDomainsValues {
+  dataLoading: boolean;
+  domains: CrawlerDomain[];
+  meta: Meta;
+}
+
+interface CrawlerDomainsResponse {
+  results: CrawlerDomainFromServer[];
+  meta: Meta;
+}
+
+interface CrawlerDomainsActions {
+  deleteDomain(domain: CrawlerDomain): { domain: CrawlerDomain };
+  fetchCrawlerDomainsData(): void;
+  onPaginate(newPageIndex: number): { newPageIndex: number };
+  onReceiveData(domains: CrawlerDomain[], meta: Meta): { domains: CrawlerDomain[]; meta: Meta };
+  crawlerDomainDeleted(data: CrawlerData): { data: CrawlerData };
+}
+
+export const CrawlerDomainsLogic = kea<MakeLogicType<CrawlerDomainsValues, CrawlerDomainsActions>>({
+  path: ['enterprise_search', 'app_search', 'crawler', 'crawler_domains_logic'],
+  actions: {
+    deleteDomain: (domain) => ({ domain }),
+    fetchCrawlerDomainsData: true,
+    onReceiveData: (domains, meta) => ({ domains, meta }),
+    onPaginate: (newPageIndex) => ({ newPageIndex }),
+    crawlerDomainDeleted: (data) => ({ data }),
```

Comment:
Here's an example of an action that publishes solely so that other views can listen for it. In this case, `CrawlerLogic`, which is a higher level logic for all Crawler pages, can listen for this and respond if any respond if a domain is deleted.